### DISCUSSION TITLE
Why does szurubooru and pretty much every booru display post counts for tags?

### DISCUSSION THREAD

**Discussion by skybldev** [rr-/szurubooru#550] [General]
One would find it almost trivial or useless. What even is the justification for it? Personally my stance on it is:
PROS:
- shows how popular a tag is
- attracts users to specific artists, especially when they have a lot to offer based on the post count for their corresponding tag
- helps with narrowing down potential query, especially if you are looking for a specific image

CONS:
- for most servers this requires querying for matching posts and counting the result, which introduces a performance impact
- (in some software) little but existent database size increase due to storing post count per tag
- a little more data sent to the client on every tag view request

Asking because a friend and I are having a deep and *i n t e l l e c t u a l* discussion about a database design.

**Comment by parallax4** [comment]
If the pros are all usability advantages and the cons are all (very minor) technical costs, it seems clear which side is more convincing.

Why didn't you list "trivial/useless/distracting/an eye sore" as a con if you included that in the first line?

**Reply by skybldev** [reply]
It is already clear to me which side is more convincing and ultimately what is a better choice. Also I didn't include "distracting" or "an eye sore", I only mentioned "trivial" and "useless" because these are the sum of the pros and cons; it is or is not proven to be so when you weigh them. 

I am more or less asking if there are any other pros or cons that people have come up with in their experiences with boorus, so that our opinions can take into account a broader set of other opinions.
